File: JAVA/GestionUtilisateur/src/main/java/org/example/model/Employee.java
package org.example.model;

/**
 * Entité Employee représentant un employé du système RH
 * Chaque employé est rattaché à un RH qui l'a créé
 */
public class Employee {
    private Integer id;
    private String firstName;
    private String lastName;
    private Integer age;
    private String jobTitle;
    private String email;
    private String password;
    private Integer rhId; // ID du RH qui a créé cet employé

    // Constructeurs
    public Employee() {
    }

    public Employee(String firstName, String lastName, Integer age, String jobTitle, 
                    String email, String password, Integer rhId) {
        this.firstName = firstName;
        this.lastName = lastName;
        this.age = age;
        this.jobTitle = jobTitle;
        this.email = email;
        this.password = password;
        this.rhId = rhId;
    }

    public Employee(Integer id, String firstName, String lastName, Integer age, 
                    String jobTitle, String email, String password, Integer rhId) {
        this.id = id;
        this.firstName = firstName;
        this.lastName = lastName;
        this.age = age;
        this.jobTitle = jobTitle;
        this.email = email;
        this.password = password;
        this.rhId = rhId;
    }

    // Getters et Setters
    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public Integer getAge() {
        return age;
    }

    public void setAge(Integer age) {
        this.age = age;
    }

    public String getJobTitle() {
        return jobTitle;
    }

    public void setJobTitle(String jobTitle) {
        this.jobTitle = jobTitle;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public Integer getRhId() {
        return rhId;
    }

    public void setRhId(Integer rhId) {
        this.rhId = rhId;
    }

    // Méthode utilitaire
    public String getFullName() {
        return firstName + " " + lastName;
    }

    @Override
    public String toString() {
        return "Employee{" +
                "id=" + id +
                ", firstName='" + firstName + '\'' +
                ", lastName='" + lastName + '\'' +
                ", age=" + age +
                ", jobTitle='" + jobTitle + '\'' +
                ", email='" + email + '\'' +
                ", rhId=" + rhId +
                '}';
    }
}

File: src/Entity/Rh/Employee.php
<?php

namespace App\Entity\Rh;

use App\Repository\Rh\EmployeeRepository;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Entity(repositoryClass: EmployeeRepository::class)]
#[ORM\Table(name: 'employees')]
#[ORM\Index(name: 'idx_rh_id', columns: ['rh_id'])]
#[ORM\Index(name: 'idx_email', columns: ['email'])]
#[ORM\HasLifecycleCallbacks]
class Employee
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(name: 'first_name', length: 100)]
    #[Assert\NotBlank(message: 'First name is required')]
    #[Assert\Length(max: 100, maxMessage: 'First name cannot exceed 100 characters')]
    private ?string $firstName = null;

    #[ORM\Column(name: 'last_name', length: 100)]
    #[Assert\NotBlank(message: 'Last name is required')]
    #[Assert\Length(max: 100, maxMessage: 'Last name cannot exceed 100 characters')]
    private ?string $lastName = null;

    #[ORM\Column]
    #[Assert\NotNull(message: 'Age is required')]
    #[Assert\GreaterThan(value: 17, message: 'Age must be at least 18')]
    #[Assert\LessThan(value: 100, message: 'Age must be less than 100')]
    private ?int $age = null;

    #[ORM\Column(name: 'job_title', length: 100)]
    #[Assert\NotBlank(message: 'Job title is required')]
    #[Assert\Length(max: 100, maxMessage: 'Job title cannot exceed 100 characters')]
    private ?string $jobTitle = null;

    #[ORM\Column(length: 100)]
    #[Assert\NotBlank(message: 'Email is required')]
    #[Assert\Email(message: 'Please enter a valid email')]
    #[Assert\Length(max: 100, maxMessage: 'Email cannot exceed 100 characters')]
    private ?string $email = null;

    #[ORM\Column(length: 255)]
    private ?string $password = null;

    #[ORM\Column(name: 'rh_id', type: Types::INTEGER)]
    #[Assert\NotNull(message: 'RH ID is required')]
    private ?int $rhId = null;

    #[ORM\Column(name: 'created_at', type: Types::DATETIME_MUTABLE, nullable: true)]
    private ?\DateTimeInterface $createdAt = null;

    #[ORM\Column(name: 'updated_at', type: Types::DATETIME_MUTABLE, nullable: true)]
    private ?\DateTimeInterface $updatedAt = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getFirstName(): ?string
    {
        return $this->firstName;
    }

    public function setFirstName(string $firstName): static
    {
        $this->firstName = $firstName;
        return $this;
    }

    public function getLastName(): ?string
    {
        return $this->lastName;
    }

    public function setLastName(string $lastName): static
    {
        $this->lastName = $lastName;
        return $this;
    }

    public function getFullName(): string
    {
        return $this->firstName . ' ' . $this->lastName;
    }

    public function getAge(): ?int
    {
        return $this->age;
    }

    public function setAge(int $age): static
    {
        $this->age = $age;
        return $this;
    }

    public function getJobTitle(): ?string
    {
        return $this->jobTitle;
    }

    public function setJobTitle(string $jobTitle): static
    {
        $this->jobTitle = $jobTitle;
        return $this;
    }

    public function getEmail(): ?string
    {
        return $this->email;
    }

    public function setEmail(string $email): static
    {
        $this->email = $email;
        return $this;
    }

    public function getPassword(): ?string
    {
        return $this->password;
    }

    public function setPassword(string $password): static
    {
        $this->password = $password;
        return $this;
    }

    public function getRhId(): ?int
    {
        return $this->rhId;
    }

    public function setRhId(int $rhId): static
    {
        $this->rhId = $rhId;
        return $this;
    }

    public function getCreatedAt(): ?\DateTimeInterface
    {
        return $this->createdAt;
    }

    public function getUpdatedAt(): ?\DateTimeInterface
    {
        return $this->updatedAt;
    }

    #[ORM\PrePersist]
    public function onPrePersist(): void
    {
        $this->createdAt = new \DateTime();
        $this->updatedAt = new \DateTime();
    }

    #[ORM\PreUpdate]
    public function onPreUpdate(): void
    {
        $this->updatedAt = new \DateTime();
    }
}
